docs/source/Plugins.rst

```diff
@@ -157,6 +157,20 @@ plugin (``json``) will be executed before the XUnit plugin (``xunit``).
 If the HTML result plugin is installed and enabled (``html``) it will
 be executed before both JSON and XUnit.
 
+Configuring the plugin execution order
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+On some circumstances it may be necessary to change the order in which plugins
+are executed.  To do so, add a ``order`` entry a configuration file section
+named after the plugin type.  For ``job.prepost`` plugin types, the section name
+has to be named ``plugins.job.prepost``, and it would look like this::
+
+  [plugins.job.prepost]
+  order = ['myplugin', 'jobscripts']
```

Comment:
The problem with this approach is that we might need different types of plugins to execute in a specific order. For example, we want the multiplexer plugin to be invoked after the remote plugin in the use-case of remote execution.

I haven't reviewed the rest of the code, but would it be possible to use something like below?

    [plugins]
    order = job.prepost.myplugin, job.prepost.jobscripts, cli.cmd.sysinfo, ...

Or even:

    [plugins]
    # plugins are loaded and run in the order they're listed below
    enabled = job.prepost.myplugin, job.prepost.jobscripts, cli.cmd.sysinfo, ...